hystrix-contrib/hystrix-javanica/hystrix-javanica-core/src/main/java/com/netflix/hystrix/contrib/javanica/command/closure/BaseClosureFactory.java

```diff
@@ -60,27 +33,29 @@ public Closure createClosure(MetaHolder metaHolder, Method method, Object o, Obj
      * @return new {@link Closure} instance
      * @throws Exception
      */
-    Closure createClosure(String rootMethodName, final Object closureObj) throws Exception {
+   protected Closure createClosure(String rootMethodName, final Object closureObj) throws Exception {
         if (!isClosureCommand(closureObj)) {
             throw new RuntimeException(format(ERROR_TYPE_MESSAGE, rootMethodName,
                     getClosureCommandType().getName()).getMessage());
         }
         Method closureMethod = closureObj.getClass().getMethod(INVOKE_METHOD);
         return new Closure(closureMethod, closureObj);
     }
+    
+    protected abstract Object execute(T metaHolder, Method method, Object o, Object... args);
```

Comment:
`execute` in Factory ? Factory is responsible only for constructing objects, ideally of course. What's the reasons of having `execute` here ?